core/Azure.Mcp.Core/tests/Azure.Mcp.Core.UnitTests/Services/Azure/BaseAzureServiceTests.cs

```diff
@@ -48,11 +48,15 @@ public async Task CreateArmClientAsync_CreatesAndUsesCachedClient()
         var client = await _azureService.GetArmClientAsync(TenantName, retryPolicyArgs);
         var client2 = await _azureService.GetArmClientAsync(TenantName, retryPolicyArgs);
 
-        Assert.Equal(client, client2);
+        Assert.NotEqual(client, client2);
 
         var otherClient = await _azureService.GetArmClientAsync(tenantName2, retryPolicyArgs);
 
         Assert.NotEqual(client, otherClient);
+
+        // Not tested: we'd like to, but can't, verify the TokenCredential is reused
+        // between client and client2 but NOT with otherClient. ArmClient doesn't expose
+        // the credential nor the HttpPipeline the credential is included within.
```

Comment:
We could add a custom pipeline policy to validate the TokenCredential reuse but that's not a blocker and can be done in a follow up PR.

Reply:
Resolving as "won't fix". Messaged offline to ask for teaching me these ways because this is something I want to do, especially after #1020.